src/components/toast/global_toast_list.test.tsx

```diff
@@ -186,5 +187,96 @@ describe('EuiGlobalToastList', () => {
         expect(dismissToastSpy).toBeCalled();
       });
     });
+
+    describe('dismissAllToasts Button', () => {
+      test('is not visible when the showClearAllButtonAt prop is set to 0', () => {
+        const dismissToastSpy = jest.fn();
+        const dismissAllToastsSpy = jest.fn();
+        const TOAST_LIFE_TIME_MS = 10;
+
+        const TOAST_COUNT = CLEAR_ALL_TOASTS_THRESHOLD_DEFAULT;
+
+        const { container } = render(
+          <EuiGlobalToastList
+            toasts={Array.from(new Array(TOAST_COUNT)).map((_, idx) => ({
+              id: String(idx),
+              'data-test-subj': String(idx),
+            }))}
+            toastLifeTimeMs={TOAST_LIFE_TIME_MS}
+            dismissToast={dismissToastSpy}
+            onClearAllToasts={dismissAllToastsSpy}
+            showClearAllButtonAt={0}
+          />
+        );
+
+        const dismissAllToastsButton = queryByTestSubject(
+          container,
+          'euiClearAllToastsButton'
+        );
+
+        expect(dismissAllToastsButton).not.toBeInTheDocument();
+      });
+
+      test('is displayed when the number of toasts to be displayed equals the internal default threshold value', async () => {
+        const dismissToastSpy = jest.fn();
+        const dismissAllToastsSpy = jest.fn();
+        const TOAST_LIFE_TIME_MS = 10;
+
+        const TOAST_COUNT = CLEAR_ALL_TOASTS_THRESHOLD_DEFAULT;
+
+        const { container } = render(
```

Comment:
Let's prefer `getByTestSubject` over `find`, so we don't have to `await`

```suggestion
        const { getByTestSubject } = render(
```